vendor/bat-native-ads/src/bat/ads/internal/flags/did_override_variations_command_line_switches/did_override_variations_command_line_switches_parser_util.cc

```diff
@@ -0,0 +1,69 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/flags/did_override_variations_command_line_switches/did_override_variations_command_line_switches_parser_util.h"
+
+#include <string>
+
+#include "base/base_switches.h"
+#include "base/check.h"
+#include "base/command_line.h"
+#include "base/containers/flat_set.h"
+#include "components/variations/variations_switches.h"
+
+namespace ads {
+
+namespace {
+
+bool HasSwitchWithValue(const std::string& switch_key) {
+  CHECK(base::CommandLine::InitializedForCurrentProcess());
+  const base::CommandLine* const command_line =
+      base::CommandLine::ForCurrentProcess();
+  return command_line->HasSwitch(switch_key) &&
+         !command_line->GetSwitchValueASCII(switch_key).empty();
+}
+
+}  // namespace
+
+bool ParseVariationsCommandLineSwitches() {
+  if (HasSwitchWithValue(variations::switches::kFakeVariationsChannel) ||
+      HasSwitchWithValue(variations::switches::kVariationsOverrideCountry)) {
+    return true;
+  } else {
+    const base::flat_set<std::string> kCommandLineSwitches = {
```

Comment:
As I understand, it is better to use a vector here, as we don't do a search through values.

Reply:
Fixed